test/livebook_web/live/session_live_test.exs

```diff
@@ -319,8 +319,12 @@ defmodule LivebookWeb.SessionLiveTest do
 
       wait_for_session_update(session.pid)
 
-      assert render(view) =~ "Updated frame"
-      refute render(view) =~ "In frame"
+      # Wait for the send_update to be applied
+      Process.sleep(10)
```

Comment:
This test covers frame update and it has been failing relatively often on CI. I believe the race condition is that LV receives the frame update message and while it's being handled, we call `render(view)` which sends another message, only then LV does `send_update` to the component, which sends message to itself. So in that scenario render is handled before the component update is applied.

Reply:
`Process.sleep(10)` is usually an anti-pattern. Typically what you do is to call a `render(view)` which you do not assert on.

So maybe:

```
# Render once so xyz is sent and processed
_ = render(view)
content = render(view)
```

Reply:
Good call, this should guarantee the correct ordering in this case! Any reason for `_ =`?

Reply:
The underscore is to make it explicitly you don’t care about the result :)

Reply:
The underscore is to make it explicitly you don’t care about the result :)